src/storage-types/src/sources.rs

```diff
@@ -2227,13 +2227,18 @@ impl RustType<ProtoPostgresSourceConnection> for PostgresSourceConnection {
 pub struct PostgresSourcePublicationDetails {
     pub tables: Vec<mz_postgres_util::desc::PostgresTableDesc>,
     pub slot: String,
+    // The active timeline_id when this source was created
+    // The None value indicates an unknown timeline, to account for sources that existed
+    // prior to this field being introduced
+    pub timeline_id: Option<u32>,
 }
 
 impl RustType<ProtoPostgresSourcePublicationDetails> for PostgresSourcePublicationDetails {
     fn into_proto(&self) -> ProtoPostgresSourcePublicationDetails {
         ProtoPostgresSourcePublicationDetails {
             tables: self.tables.iter().map(|t| t.into_proto()).collect(),
             slot: self.slot.clone(),
+            timeline_id: self.timeline_id.clone().unwrap_or_default(),
```

Comment:
I don't think `unwrap_or_default` is the right call here––you can mark the proto field to be `optional`, in which case it will accept an option here.